ElectronClient/app/gui/NoteText.jsx

```diff
@@ -1235,81 +1235,92 @@ class NoteTextComponent extends React.Component {
 	}
 
 	async printTo_(target, options) {
-		if (this.props.selectedNoteIds.length !== 1 || !this.webviewRef_.current) {
-			throw new Error(_('Only one note can be printed or exported to PDF at a time.'));
-		}
-
 		// Concurrent print calls are disallowed to avoid incorrect settings being restored upon completion
 		if (this.isPrinting_) {
+			console.log(`Printing ${options.path} to ${target} disallowed, already printing.`);
 			return;
 		}
 
 		this.isPrinting_ = true;
 
-		// const previousBody = this.state.note.body;
-		// const tempBody = `${this.state.note.title}\n\n${previousBody}`;
-
-		// const previousTheme = Setting.value('theme');
-		// Setting.setValue('theme', Setting.THEME_LIGHT);
-		// this.lastSetHtml_ = '';
-		// await this.updateHtml(this.state.note.markup_language, tempBody, { useCustomCss: true });
-		// this.forceUpdate();
-
-		// const restoreSettings = async () => {
-		// 	Setting.setValue('theme', previousTheme);
-		// 	this.lastSetHtml_ = '';
-		// 	await this.updateHtml(this.state.note.markup_language, previousBody);
-		// 	this.forceUpdate();
-		// };
-
 		// Need to save because the interop service reloads the note from the database
 		await this.saveIfNeeded();
 
-		setTimeout(async () => {
-			if (target === 'pdf') {
-				try {
-					const pdfData = await InteropServiceHelper.exportNoteToPdf(options.noteId, {
-						printBackground: true,
-						pageSize: Setting.value('export.pdfPageSize'),
-						landscape: Setting.value('export.pdfPageOrientation') === 'landscape',
-						customCss: this.props.customCss,
-					});
-					await shim.fsDriver().writeFile(options.path, pdfData, 'buffer');
-				} catch (error) {
-					console.error(error);
-					bridge().showErrorMessageBox(error.message);
-				}
-			} else if (target === 'printer') {
-				try {
-					await InteropServiceHelper.printNote(options.noteId, {
-						printBackground: true,
-						customCss: this.props.customCss,
-					});
-				} catch (error) {
-					console.error(error);
-					bridge().showErrorMessageBox(error.message);
-				}
+		await new Promise(resolve => {
+			setTimeout(resolve, 100);
+		});
```

Comment:
Maybe use time.msleep(100) from time-utils?

Reply:
It works, done!

Actually, do you know why we have this delay? On my system it works fine without it.  So I'm not sure what it is safeguarding.  Perhaps we can just remove it.

Reply:
You're right, it was a left over from when the theme was being switched before printing. In that case, we had to wait till the UI was updated before printing, but it works very differently now. So indeed it's probably not needed. If it's working for your without it, please remove it.

Reply:
Great, done.